### DISCUSSION TITLE
How to swap the combos Tab and ESC? Misc questions

### DISCUSSION THREAD

**Discussion by enoryw** [urob/zmk-config#70] [Q&A]
How to swap the combos Tab and ESC? I want ESC on hrm instead of Tab because it's used more frequently. I'm using zmk-helpers v2 and looking at combos.dtsi but am kind of lost (not a programmer)--it seems tab is implemented slightly differently than the rest of the homerow combos as I don't see an entry for it on base.keymap, unlike 
the other HRM-combos e.g.:

    MAKE_HRM(hml_repeat, &kp, &key_repeat, KEYS_R THUMBS)  // used for HRM-combo
    MAKE_HRM(hmr_lpar_lt, &kp, &lpar_lt, KEYS_L THUMBS)    // "    "   "   "
    MAKE_HRM(hmr_rpar_gt, &kp, &rpar_gt, KEYS_L THUMBS)    // "    "   "   "

I'm not swapper since I'm on Linux. 

Unrelated questions:

* For adding more layers any suggestions for how to access them (general approach for any 34-key layout)? I want to add a symbol layer--should I begin to bind them to the 4 non-alpha keys on the base layer for a 34-key layout? Exhaust all thumb combos first? 

* Would merging FN layer into NUM layer (putting F keys on right side) be potentially problematic and/or require additional tweaks? Looking to add this [symbols layer](https://getreuer.info/posts/keyboards/symbol-layer/index.html#my-symbol-layer). Not sure if there are any popular bindings that require use of mod keys (I suppose if that's needed I can still add HRMs to this layer.

* Zmk mouse keys don't seem to work on Linux and I've come across one other comment that says the same--I'm wondering if anyone can confirm this. I press w + p combo, then press and hold the nav layer then the up/down/left/right on that layer but it just acts if it's just on the nav layer. Not sure how to diagnose this further.

Thanks!

**Comment by urob** [comment]
The `TAB` combo is defined here: https://github.com/urob/zmk-config/blob/54d710faa22caad4691475c3adc3a9df7dd638e6/config/combos.dtsi#L27

Note that it is bound to `HRC_TAB`, which is defined further up the same file, in the same place where `HRC_REPEAT` etc are defined:
https://github.com/urob/zmk-config/blob/54d710faa22caad4691475c3adc3a9df7dd638e6/config/combos.dtsi#L17-L20

(The definition for `HRC_TAB` is a bit simpler than `HRC_REPEAT` because the hold-tap isn't bound to a custom behavior, so we can use `&hml` instead of having to create a custom hold-tap.)

**Comment by urob** [comment]
**Re: adding more layers.** I personally _really_ dislike lateral thumb movements and couldn't stand having something as frequently used as a symbol layer being activated by a secondary thumb. If I weren't using combos for symbols, I'd probably use hold-taps on [`LB1` and `RB1`](https://github.com/urob/zmk-helpers/blob/v2/docs/img/key_labels_example.png) to activate it.

But these things are _very_ subjective and I suggest trying out a few options to see what's most comfortable for you.

The same applies to merging `FN` and `NUM` layers. I did quite a bit of tweaking and in my experience things indeed are highly interdependent. Alas, often only trial & error will be able to tell what works best for you. 

**Re: mouse on linux.** I currently don't have access to any unix other than through WSL. There was another [comment](https://github.com/urob/zmk-config/discussions/64#discussioncomment-9978655) on here in a similar spirit. It might be more fruitful to respond to them over there to keep the discussion in one place.
